src/mpid/ch4/netmod/ofi/subconfigure.m4

```diff
@@ -134,10 +68,151 @@ AM_COND_IF([BUILD_CH4_NETMOD_OFI],[
         PAC_SET_HEADER_LIB_PATH(libfabric)
     fi
 
+    runtime_capabilities="no"
+    no_providers="no"
+    # $netmod_args - contains the OFI provider
+    if test "${netmod_args#* }" != "$netmod_args" ; then
+        runtime_capabilities="yes"
+        AC_MSG_NOTICE([Using runtime capability set due to multiple selected providers])
+    elif test "x$netmod_args" = "x" ; then
+        runtime_capabilities="yes"
+        no_providers="yes"
+        AC_MSG_NOTICE([Using runtime capability set due to no selected provider])
+        AC_DEFINE([MPIDI_CH4_OFI_USE_SET_RUNTIME], [1], [Define to use runtime capability set])
+    fi
+
+    if test "$no_providers" = "no" ; then
+        enable_psm="no"
+        enable_psm2="no"
+        enable_sockets="no"
+        enable_verbs="no"
+        enable_usnic="no"
+        enable_mxm="no"
+        enable_gni="no"
+        enable_udp="no"
+        enable_rxm="no"
+        enable_rxd="no"
+    else
+        enable_psm="yes"
+        enable_psm2="yes"
+        enable_sockets="yes"
+        enable_verbs="yes"
+        enable_usnic="yes"
+        enable_mxm="yes"
+        enable_gni="yes"
+        enable_udp="yes"
+        enable_rxm="yes"
+        enable_rxd="yes"
+    fi
+
+    for provider in $netmod_args ; do
```

Comment:
It'll be good to reorder this to two categories:

1. Providers where we know the exact capabilities.

2. Providers where we do not know the exact capabilities (where we have to use runtime detection).

Can you push the second category down the list and add a comment saying so?